### ISSUE TITLE
Allow selection of effect units as quick effects

### ISSUE THREAD

**Issue by chris-echoz** [mixxxdj/mixxx#10962]
### Feature Description

Having the option to let the quick effect knob control and apply effect chains from effect units would be very useful for certain controllers like the Native Instruments Traktor Kontrol S2 MK3 (and likely most other NI controllers) that have very limited controls for effects:
![image](https://user-images.githubusercontent.com/7262891/195861439-653b6eae-8f35-4b29-b368-55128fbbea2d.png)

In 2.3 these are not very useful at all, but with #4467 (to my understanding, based on skimming through #2618) these buttons can be mapped to set the quick effect to a given index in the effect chain preset list.

Ideally what I'm thinking is that the effect units can be included, excluded and reordered just like any other effect chain in the settings. Then when selected, rather than using a static preset, it would use the same effect chain as the one in the effect unit so that you can select an effect unit to be applied as a quick effect with the drop-down (or the FX select buttons in the picture), control the meta-knob with the quick effects knob (mapped to the two knobs in the picture), but also easily use the more extensive controls of the effect unit with the mouse/touch without having to do anything in the options or deal with saving and loading of presets.

Looking at the code it's clear that it's not as straightforward as just adding the effect units' chains to the list, as the quick effect requires an effect chain preset, not an effect chain, but I'm not familiar enough with the code base to even know whether or not something like this would be feasible in some other way.

I'd love to hear what someone familiar with the refactored effects system thinks about this.

**Comment by ronso0** [comment]
The difference between QuickEffect and standard chains is that QuickEffect chains have **Mix** knob (dry/wet) always set to fully wet. (not sure how the mix mode toggle is handled)
That said, should picking a standard chain in a QuickEffect chain set that to fully wet, too?

So, should standard chains in the QuickEffect preset list simply be shortcuts/combo toggles for 
* `[EffectRack1_EffectUnit1],group_[ChannelN]_enable` and
* `[EffectRack1_EffectUnit1],mix` > `1`

and `[QuickEffectRack1_[ChannelN]],super1` becomes a proxy for
`[EffectRack1_EffectUnit1],super1`?

**Comment by chris-echoz** [comment]
@ronso0 Yeah, that's almost exactly what I'm thinking in terms of behavior, except that each quick effect knob would still only apply to its own channel. I realize now this may pose a challenge since directly proxying the quick effect knob would affect all channels.

I think the most user-friendly way to solve that would probably be to let the standard effect chain meta knobs set the max value and then the quick effect knob only works within that range, so as an example you you may have FX1 set up with echo at 50%, and flanger at 25%, turning the quick effect knob to 50% would apply 25% echo and 12.5% flanger.

This does probably pose a challenge for effects where the center is zero, and I'm not sure how that would be solved or even how it's currently solved if you have an effect chain with, say, echo and filter and assign it as a quick effect.

**Comment by Be-ing** [comment]
> The difference between QuickEffect and standard chains is that QuickEffect chains have Mix knob (dry/wet) always set to fully wet. (not sure how the mix mode toggle is handled)

That's correct, and the other very big difference is that the QuickEffects are not reroutable; there is a 1:1 association between a QuickEffect chain and a deck.

I don't think this could work quite the way you've proposed, but I think I understand what you're getting at. The workflow for transfering state between the regular effect chains and the QuickEffects could be streamlined. What could be nice to add is a means to save the current state of a regular effect chain as a preset and immediately load it into a different chain, kinda like the clone from deck feature does with a deck's state.

**Comment by chris-echoz** [comment]
@Be-ing I suppose one possible approach to effectively achieve what I proposed might be to save a preset on any event where parameters were changed? Not sure how the quick effects deal with reloading a changed preset and if that might also reset the knob, but I'll try to have a look at that when I have time.
